tests/validation/cattlevalidationtest/core/test_native_docker.py

```diff
@@ -0,0 +1,198 @@
+
+TEST_IMAGE = 'ibuildthecloud/helloworld'
+
+socat_test_image = os.environ.get('CATTLE_CLUSTER_SOCAT_IMAGE',
+                                  'docker:sonchang/socat-test')
+
+
+def _native_doc_check():
+    return os.environ.get('DOCKER_HOST') is None or os.environ.get(
+        'DOCKER_TEST') == 'false'
+
+if_native_docker = pytest.mark.skipif(_native_doc_check(),
+                                      reason='Environment not configured for '
+                                             'native docker tests.')
+
+
+@pytest.fixture(scope='module')
+def docker_client(client, unmanaged_network, request):
+    # When these tests run in the CI environment, the hosts don't expose the
+    # docker daemon over tcp, so we need to create a container that binds to
+    # the docker socket and exposes it on a port
+    hosts = client.list_host(kind='docker', removed_null=True)
+    assert len(hosts) >= 1
+    host = hosts[0]
+    socat_container = client.create_container(
+        name='socat-%s' % random_str(),
+        networkIds=[unmanaged_network.id],
+        imageUuid=socat_test_image,
+        ports='2375',
+        stdinOpen=False,
+        tty=False,
+        publishAllPorts=True,
+        dataVolumes='/var/run/docker.sock:/var/run/docker.sock',
+        requestedHostId=host.id)
+
+    def remove_socat():
+        client.delete(socat_container)
+
+    request.addfinalizer(remove_socat)
+
+    wait_for_condition(
+        client, socat_container,
+        lambda x: x.state == 'running',
+        lambda x: 'State is: ' + x.state)
+
+    socat_container = client.reload(socat_container)
+    ip = host.ipAddresses()[0].address
+    port = socat_container.ports()[0].publicPort
+
+    params = {}
+    params['base_url'] = 'tcp://%s:%s' % (ip, port)
+    api_version = os.getenv('DOCKER_API_VERSION', '1.15')
+    params['version'] = api_version
+
+    return Client(**params)
+
+
+@pytest.fixture(scope='module')
+def pull_images(docker_client):
+    image = (TEST_IMAGE, 'latest')
+    images = docker_client.images(image[0])
+    if not images:
+        docker_client.pull(image[0], image[1])
+
+
+@if_native_docker
+def test_native_unmanaged_network(docker_client, admin_client, pull_images):
+    name = 'native-%s' % random_str()
+    d_container = docker_client.create_container(TEST_IMAGE,
+                                                 name=name)
+    docker_client.start(d_container)
+    inspect = docker_client.inspect_container(d_container)
+
+    def check():
+        containers = admin_client.list_container(name=name)
+        return len(containers) > 0
+
+    wait_for(check, timeout_message=CONTAINER_APPEAR_TIMEOUT_MSG % name)
+
+    r_containers = admin_client.list_container(name=name)
+    assert len(r_containers) == 1
+    c = r_containers[0]
+    c = admin_client.wait_success(c)
+
+    assert c.externalId == d_container['Id']
+    assert c.state == 'running'
+    assert c.primaryIpAddress == inspect['NetworkSettings']['IPAddress']
+
```

Comment:
Might be an overkill, but can we also check that container volume(s) was propagated to cattle?